src/realitycheck/realitycheck.js

```diff
@@ -98,6 +96,11 @@ define(["jquery", "windows/windows", "websockets/binary_websockets", "lodash", '
 
                                 require(['text!realitycheck/realitycheck.html', "css!realitycheck/realitycheck.css"], function(html) {
                                     var div = $(html).i18n();
+                                    // Is there a better option?
+                                    if(win){
+                                        resolve();
+                                        return;
+                                    }
                                     win = windows.createBlankWindow($('<div/>'), {
```

Comment:
@aminroosta The best way to implement this is to use combination of promise and event manager. You can get example from windows.js/binary_sockets.js how @aminroosta did it. What you can do is, wherever init() is being called from, return a promise(which we are already doing). Then resolve all these promises when "ready" event is emitted.